src/components/OSCALMetadata.js

```diff
@@ -38,17 +53,209 @@ const OSCALMetadataPartiesCard = styled(Paper)(({ theme }) => ({
+  if (!mapping || !contactType) {
+    return UNKNOWN_TYPE_ICON;
+  }
+  return mapping[contactType] ?? UNKNOWN_TYPE_ICON;
+}
+
+function getPhoneIcon(contactType) {
+  return getIconOfType(TELEPHONE_ICON_MAPPING, contactType);
+}
+
+function getAddressIcon(contactType) {
+  return getIconOfType(ADDRESS_ICON_MAPPING, contactType);
+}
+
+export function OSCALMetadataPartyAddress(props) {
+  const { address } = props;
+  const addrString = [
+    ...address["addr-lines"],
+    `${address.city}, ${address.state} ${address["postal-code"]}`,
+    address.country,
+  ]
+    .filter((line) => line)
+    .flatMap((item) => [item, <br key={item} />]);
+
+  return (
+    <Grid container spacing={1}>
+      <Grid item>{getAddressIcon(address.type)}</Grid>
+      <Grid item>
+        <Typography>{addrString}</Typography>
+      </Grid>
+    </Grid>
+  );
+}
+
+export function OSCALMetadataPartyEmail(props) {
+  return (
+    <Typography>
+      <Link href={`mailto:${props.email}`}>{props.email}</Link>
+    </Typography>
+  );
+}
+
+export function OSCALMetadataPartyTelephone(props) {
+  const { telephone } = props;
+  return (
+    <Typography>
+      {getPhoneIcon(telephone.type)}
+      <Link href={`tel:${telephone.number}`}>{telephone.number}</Link>
+    </Typography>
+  );
+}
+
+export function OSCALMetadataPartyDialog(props) {
+  const [open, setOpen] = React.useState(false);
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const TYPE_MAPPING = (infoProps) => ({
+    address: <OSCALMetadataPartyAddress address={infoProps} />,
+    telephone: <OSCALMetadataPartyTelephone telephone={infoProps} />,
+    email: <OSCALMetadataPartyEmail email={infoProps} />,
+  });
+
+  const getPartyInfoList = (
+    list,
+    partyInfoType,
+    emptyMessage = "No information provided"
+  ) => {
+    if (!list?.length) {
+      return <Typography> {emptyMessage} </Typography>;
+    }
+
+    return list.map((item) => (
+      <ListItem
+        key={partyInfoType === "email" ? item : `${item?.type}--${item?.name}`}
+      >
+        {TYPE_MAPPING(item)[partyInfoType]}
+      </ListItem>
+    ));
+  };
+
+  return (
+    <CardActions>
+      <Button size="small" onClick={handleOpen}>
+        Contact
+      </Button>
+      <Dialog
+        open={open}
+        onClose={handleClose}
+        scroll="paper"
+        aria-labelledby="scroll-dialog-title"
+        aria-describedby="scroll-dialog-description"
+        maxWidth="md"
+        fullWidth
+      >
+        <DialogTitle id="scroll-dialog-title">{props.party.name}</DialogTitle>
+        <DialogContent>
+          <Grid container spacing={1}>
+            <Grid item xs={4}>
+              <Typography variant="h6" component="h3">
+                <MapIcon />
+                Addresses
+              </Typography>
+              <List>
+                {getPartyInfoList(
+                  props.party.addresses,
+                  "address",
```

Comment:
Can you find a way to use an enum for this (and email and telephone when used in the same context)?